main.go

```diff
@@ -75,6 +75,16 @@ func handler(w http.ResponseWriter, r *http.Request, logger log.Logger) {
 		return
 	}
 
+	authName := query.Get("auth")
+	if len(query["auth"]) > 1 {
+		http.Error(w, "'auth' parameter must only be specified once", 400)
```

Comment:
Whilst we're touching this code, it might be a nice opportunity to replace the numeric HTTP status codes with package constants from net/http, e.g. `http.StatusBadRequest`.